src/kudu/consensus/raft_consensus.cc

```diff
@@ -1248,6 +1249,18 @@ static bool IsConsensusOnlyOperation(OperationType op_type) {
 Status RaftConsensus::StartFollowerTransactionUnlocked(const ReplicateRefPtr& msg) {
   DCHECK(lock_.is_locked());
 
+  // Validate crc32 checksum
+  uint32_t payload_crc32 = msg->get()->write_payload().crc32();
+  if (payload_crc32 != 0) {
+    const std::string& payload = msg->get()->write_payload().payload();
+    uint32_t computed_crc32 = crc::Crc32c(payload.c_str(), payload.size());
+    if (payload_crc32 != computed_crc32) {
+      std::string err_msg = Substitute("Rejected: Payload corruption for $0",
+          OpIdToString(msg->get()->id()));
+      return Status::Corruption(err_msg);
```

Comment:
Can we do something more here. Can we send an information packet to MySQL Raft that
a corruption has been encountered, or we can store it within RaftConsensus as well.

In the future we can then stall or atleast introspect using the index.

Reply:
discussed offline with @bhatvinay that we will do this followup in a future PR.

Reply:
The general idea would be to have a generic framework through which kuduraft can report errors to upper layers (the plugin in our case).
1. Could be a virtual method in ConsensusRoundHandler (maybe `ReportError(...)`) which gets implemented by MysqlRaft
2. A lambda captured through options which can be invoked by kuduraft to report errors to upper layers (like a callback)

Will be done as a followup.